src/app/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import * as S from 'app/components/Modal/Modal.style';
+
+const Modal = ({
+    children,
+    visible,
+    setVisible,
+}: {
+    children: React.ReactNode;
+    visible: boolean;
+    setVisible: Dispatch<SetStateAction<boolean>>;
+}) => {
+    const handleClose = () => setVisible(false);
+
+    return (
+        <S.ModalOverlay className="modal" visible={visible} onClick={handleClose}>
+            <S.ModalWrapper onClick={e => e.stopPropagation()}>
```

Comment:
e.stopPropagation()  - it is better to put this expression in a separate function, for example stopProgationHandler, if you really need to stop the propagation in this case.